qiskit_quantuminspire/qi_results.py

```diff
@@ -1,43 +1,72 @@
-from typing import List
-
-from compute_api_client import Result as RawJobResult
 from qiskit.providers import JobV1 as Job
+from qiskit.qobj import QobjExperimentHeader
 from qiskit.result.models import ExperimentResult, ExperimentResultData
 from qiskit.result.result import Result
 
 
 class QIResult:
     """Handle QuantumInspire (batch job) results to integrate with Qiskit's Result interface."""
 
-    def __init__(self, qinspire_results: List[RawJobResult]) -> None:
-        self._raw_results = qinspire_results
-
-    def process(self, job: Job) -> Result:
-        """Process the raw job results obtained from QuantumInspire.
+    def __init__(self, job: Job) -> None:
+        """Initialize the result processor for QuantumInspire job results.
 
         Args:
             job: The (batch) job for which the results were obtained. While specified as `Job`
-            to avoid circular dependency, it is a `QIJob`.
+                to avoid circular dependency, it is a `QIJob`.
         Returns:
-            The processed results as a Qiskit Result.
+            None.
         """
 
+        self._job = job  # The batch job
+
+    def process(self) -> Result:
```

Comment:
@Mythir Wondering if this should indeed be a class now as it only has a single method. Shall we make it a function instead ?

Reply:
Good question. The only thing this does is change our `Result` into a Qiskit `Result`. So maybe this should just be part of qi_jobs.py?

Reply:
I moved it to QIJobs class